File: 백준/Gold/27297. 맨해튼에서의 모임/맨해튼에서의 모임.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.*;
public class Main{
	
	public static void main (String [] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		
		StringTokenizer st = new StringTokenizer(br.readLine());
		int N = Integer.parseInt(st.nextToken());
		int M = Integer.parseInt(st.nextToken());
		long [][] arr = new long [N][M];

		
		for (int i = 0; i < M; i++) {
	        String[] st1 = br.readLine().split(" ");
			for (int j = 0; j < N; j++) {
				long s = Long.parseLong(st1[j]);
				arr[j][i] = s;
			}
		}
        
		for (int i = 0; i < N; i++) {
			Arrays.sort(arr[i]);
		}
        
		int middle = 0;
		if(N%2 == 0) {
			middle = M /2;
		}else {
			middle = (M-1)/2;
		}
        
		long [] dot = new long [N];
		long ans = 0;
		for (int i = 0; i < N; i++) {
			dot[i] = arr[i][middle];
			for (int j = 0; j < M; j++) {
				ans += Math.abs(arr[i][middle] - arr[i][j]);
			}
		}
        
		System.out.println(ans);
		for (int i = 0; i < dot.length; i++) {
			System.out.print(dot[i]+" ");
		}
	}
}

File: 백준/Gold/27297. 맨해튼에서의 모임/맨해튼에서의 모임.py
import sys
input = sys.stdin.readline

N, M = map(int, input().split())
arr = [[0 for _ in range(M)] for _ in range(N)]
for i in range(M):
    dot = list(map(int, input().split()))
    for j in range(N):
        arr[j][i] = dot[j]

for i in range(N):
    arr[i].sort()

if M % 2:
    middle = (M - 1) // 2
else:
    middle = M // 2

total = 0
ans = []
for i in range(N):
    spot = arr[i][middle]
    ans.append(spot)
    for j in range(M):
        total += abs(spot - arr[i][j])

print(total)
print(*ans)